### DISCUSSION TITLE
图层删除问题

### DISCUSSION THREAD

**Discussion by ZhangWei-XG** [Tencent/libpag#2270] [General] [answered]
我在使用 removeLayerAt 删除下标为0 图层的时候，会把整个背景都删掉了，但是在PAGViewer 里面看到的第0位置是个小图标，请问怎么解决？

**Comment by kevingpqi123** [comment, marked as answer]
removeLayerAt 针对的是 PAGFile 或 PAGComposition 中的所有图层， PAGViewer 中你看到的有文本编辑和图片编辑两项，这里的 index 只针对 PAGFile 或 PAGComposition 中的文本图层和图片图层，使用的是下面的接口
Return an array of layers by specified editable index and layer type.

- (NSArray<PAGLayer*> *)getLayersByEditableIndex:(int)index layerType:(PAGLayerType)type
Declared In
PAGFile.h
